bootstrap/api/v1alpha1/rke2config_types.go

```diff
@@ -17,48 +17,593 @@ limitations under the License.
+
+	// PostRKE2Commands specifies extra commands to run after rke2 setup runs.
+	//+optional
+	PostRKE2Commands []string `json:"postRKE2Commands,omitempty"`
+
+	// AgentConfig specifies configuration for the agent nodes.
+	//+optional
+	AgentConfig RKE2AgentConfig `json:"agentConfig,omitempty"`
+
+	// ServerConfig specifies configuration for the agent nodes.
+	//+optional
+	ServerConfig RKE2ServerConfig `json:"serverConfig,omitempty"`
+
+	// PrivateRegistriesConfig defines the containerd configuration for private registries and local registry mirrors.
+	//+optional
+	PrivateRegistriesConfig Registry `json:"privateRegistriesConfig,omitempty"`
+
+	// Version specifies the rke2 version.
+	//+optional
+	Version string `json:"version,omitempty"`
+}
+
+type RKE2ServerConfig struct {
+	// BindAddress describes the rke2 bind address (default: 0.0.0.0).
+	//+optional
+	BindAddress string `json:"bindAddress,omitempty"`
+
+	// AdvertiseAddress IP address that apiserver uses to advertise to members of the cluster (default: node-external-ip/node-ip).
+	//+optional
+	AdvertiseAddress string `json:"advertiseAddress,omitempty"`
+
+	// TLSSan Add additional hostname or IP as a Subject Alternative Name in the TLS cert.
+	//+optional
+	TLSSan []string `json:"tlsSan,omitempty"`
+
+	// DataDir is the Folder to hold RKE2's state (default: "/var/lib/rancher/rke2").
+	//+optional
+	DataDir string `json:"data-dir,omitempty"`
+
+	// ServiceNodePortRange is the port range to reserve for services with NodePort visibility (default: "30000-32767").
+	//+optional
+	ServiceNodePortRange string `json:"service-node-port-range,omitempty"`
+
+	// ClusterDNS  Cluster IP for coredns service. Should be in your service-cidr range (default: 10.43.0.10).
+	//+optional
+	ClusterDNS string `json:"clusterDNS,omitempty"`
+
+	// ClusterDomain Cluster Domain (default: "cluster.local").
+	//+optional
+	ClusterDomain string `json:"clusterDomain,omitempty"`
+
+	// WriteKubeconfig gives the path to which kubeconfig file for admin client will be written.
+	//+optional
+	WriteKubeconfig string `json:"writeKubeconfig,omitempty"`
+
+	// WriteKubeconfigMode Writes kubeconfig with this mode.
+	//+optional
+	WriteKubeconfigMode string `json:"writeKubeconfigMode,omitempty"`
+
+	// EtcdExposeMetrics Expose etcd metrics to client interface. (Default false).
+	//+optional
+	EtcdExposeMetrics string `json:"etcdExposeMetrics,omitempty"`
+
+	// EtcdBackupConfig defines how RKE2 will snapshot ETCD: target storage, schedule, etc.
+	//+optional
+	EtcdBackupConfig EtcdBackupConfig `json:"etcdBackupConfig,omitempty"`
+
+	// Disable Do not deploy packaged components and delete any deployed components (valid items: rke2-coredns, rke2-ingress-nginx, rke2-metrics-server).
+	//+optional
+	Disable []DisabledItem `json:"disable,omitempty"`
+
+	// DisableScheduler Disable Kubernetes default scheduler.
+	//+optional
+	DisableScheduler string `json:"disable-scheduler,omitempty"`
```

Comment:
With this, `DisableCloudController`, `DisableKubeProxy` we can add them to their respective structs.